example/lib/pages/post_detail/post_detail_page_scenes.dart

```diff
@@ -1,13 +1,22 @@
 import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
 import 'package:stager/stager.dart';
 
 import '../../shared/post.dart';
 import 'post_detail_page.dart';
 
 /// A scene demonstrating a [PostDetailPage] with content.
 class PostDetailPageScene extends Scene {
-  /// The post being previewed.
-  Post currentPost = Post.fakePosts().first;
+  static const String _currentPostKey = 'currentPost';
+
+  @override
+  Future<void> setUp(EnvironmentState environmentState) async {
+    super.setUp(environmentState);
+    environmentState.setDefault(
```

Comment:
This seems to add another distributed place where you need to configure the environment control. I do wonder if you had a central place to configure each environment control if it'd be a bit easier to understand what was going on and easier to understand. E.g. rather than the instructions being: "for each environment control, define a key in the environment state, set the default in 'setUp', return the control from environmentControlBuilders, and index the state based on the key from build()" it could be more like:
"Create an environment control using the constructor (which has all the required params like default value etc) and then access the value from build()."

Reply:
I'm all for making this simpler/easier to wrap one's head around, although I'm not totally sure what a good API for that would look like. This assumes a 1:1 mapping of controls to environment variables, which isn't currently a requirement, but is possibly/probably a safe assumption?

Sketching something out:

```dart
abstract class EnvironmentControl<T> {
  EnvironmentControl({
    required this.stateKey,
    required this.defaultValue,
  });

  final String stateKey;

  final T defaultValue;

  // [context] because this is a build function.
  // [currentValue] for convenience (don't need to fetch it from [state]).
  // [state] so the widget built here can update the value.
  Widget build(BuildContext context, T currentValue, EnvironmentState state);
}
```

Does this seem reasonable/like what you had in mind?

Reply:
Took a whack at this and overall like it much better. Check out https://github.com/bryanoltman/stager/pull/41 and let me know what you think.